Automattic/woocommerce-payments
# Adds subscription product service

**Important:** This PR relies on another [WIP PR in WCPay Server](1005-gh-Automattic/woocommerce-payments-server)

See pb0GrA-1g9-p2

#### Changes proposed in this Pull Request

This adds a subscription product service that handles syncing WC subscription products with Stripe. When new subscription products are created or updated in WP-Admin, they are also created in the Stripe connect account.

#### Testing instructions

- In WP-Admin create a simple subscription product
- From the connected Stripe account, verify this was created properly:
  - Name
  - Description
  - Price
  - Interval
  - Period
  - Currency
- Update the simple subscription product
- From the connected Stripe account, verify this was updated properly
- Trash the simple subscription product in WC
- From the connected Stripe account, verify the product AND price have been archived
- Untrash the simple subscription product in WC
- From the connected Stripe account, verify the product AND price have been unarchived
- Follow the above steps for a variable subscription product, this time checking for all variations in Stripe

-------------------
